public/app/features/manage-dashboards/components/ImportDashboardForm.tsx

```diff
@@ -0,0 +1,185 @@
+  uidExists: boolean;
+  uidError: string;
+  nameExists: boolean;
+  nameError: string;
+  meta?: any;
+
+  resetDashboard: typeof resetDashboard;
+  changeDashboardTitle: typeof changeDashboardTitle;
+  saveDashboard: typeof saveDashboard;
+  changeDashboardUid: typeof changeDashboardUid;
+}
+
+interface State {
+  folderId: number;
+  uidReset: boolean;
+}
+
+class ImportDashboardForm extends PureComponent<Props, State> {
+  state: State = {
+    folderId: 0,
+    uidReset: false,
+  };
+
+  onSubmit = () => {
+    this.props.saveDashboard(this.state.folderId);
+  };
+
+  onCancel = () => {
+    this.props.resetDashboard();
+  };
+
+  onTitleChange = (event: FormEvent<HTMLInputElement>) => {
+    this.props.changeDashboardTitle(event.currentTarget.value);
+  };
+
+  onFolderChange = ($folder: { title: string; id: number }) => {
+    this.setState({ folderId: $folder.id });
+  };
+
+  onUidChange = (event: FormEvent<HTMLInputElement>) => {
+    this.props.changeDashboardUid(event.currentTarget.value);
+  };
+
+  onUidReset = () => {
+    this.setState({ uidReset: true });
+  };
+
+  render() {
+    const { dashboard, inputs, meta, source, uidExists, uidError, nameExists, nameError } = this.props;
+    const { uidReset } = this.state;
+
+    const buttonVariant: ButtonVariant = uidExists || nameExists ? 'destructive' : 'primary';
+    const buttonText = uidExists || nameExists ? 'Import (Overwrite)' : 'Import';
+
+    return (
+      <>
+        {source === DashboardSource.Gcom && (
+          <div style={{ marginBottom: '24px' }}>
+            <div>
+              <Forms.Legend>
+                Importing Dashboard from{' '}
+                <a
+                  href={`https://grafana.com/dashboards/${dashboard.gnetId}`}
+                  className="external-link"
+                  target="_blank"
+                >
+                  Grafana.com
+                </a>
+              </Forms.Legend>
+            </div>
+            <table className="filter-table form-inline">
+              <tbody>
+                <tr>
+                  <td>Published by</td>
+                  <td>{meta.orgName}</td>
+                </tr>
+                <tr>
+                  <td>Updated on</td>
+                  <td>{dateTime(meta.updatedAt).format('YYYY-MM-DD HH:mm:ss')}</td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+        )}
+        <Forms.Form onSubmit={this.onSubmit}>
+          {() => {
+            return (
+              <>
+                <Forms.Legend className="section-heading">Options</Forms.Legend>
+                <Forms.Field label="Name" invalid={nameExists} error={nameError}>
+                  <Forms.Input size="md" type="text" value={dashboard.title} onChange={this.onTitleChange} />
+                </Forms.Field>
+                <Forms.Field label="Folder">
+                  <FolderPicker onChange={this.onFolderChange} useInNextGenForms={true} initialFolderId={0} />
```

Comment:
See how to use FolderPicker with Form component here: https://github.com/grafana/grafana/pull/22395/files#diff-d2936ffa0c7be9e829013ec0d9f451a2R81